backend/devices/bitbox/relay/channel.go

```diff
@@ -148,13 +148,13 @@ func (channel *Channel) waitForValue(duration time.Duration, name string) (strin
 			unlock()
 			message, err := PullOldestMessage(relayServer(), channel)
 			if err != nil {
-				return "", err
+				return "", errp.New("pullFailed")
```

Comment:
normally you shouldn't match error text, as error text should be descriptive and should be able to change without breaking anything.

the usual solution is to return a global `var ErrPullFailed = errors.New("<descriptive error>")` and return this instead, and compare against this.

I'm ok skipping this fix however, up to you.

Reply:
Introducing a global seems like a bad solution as well. How about a function returning the error given const string arguments?

Reply:
Pushed a commit to change to constant error values. Please review.

Reply:
seems like it, but it is not ;) `Err... error` global vars should be treated as immutable, and immutable globals are not evil as opposed to mutable ones. you would mark it as `const` if Go would let you, but it does not :(

anyway, merging!

Reply:
To elaborate; it's a normal Go pattern, also used in the stdlib, for example the list of io errors:

https://golang.org/pkg/io/#pkg-variables

Reply:
Huh, so structs with no public members are counted as immutable?

Reply:
the struct is not mutable, but the global variable is.

```Go
somepackage.ErrShouldBeImmutable = errors.New("lol i changed you")
```

too bad Go does not let you mark more stuff as const.

Reply:
Example:

```Go

package main

import (
	"bytes"
	"errors"
	"fmt"
	"io"
)

func main() {
	io.EOF = errors.New("this is not the string you were looking for")
	buf := bytes.NewReader(nil)
	_, err := buf.ReadByte()
	fmt.Println(err)
}
```

```
go run main.go
this is not the string you were looking for
```

Reply:
Alternatively, another way some advocate is to use interfaces. For instance: 
https://play.golang.org/p/uTa56pGLcBC

But these days probably the best is to use the new Is and As helpers from the stdlib errors package https://golang.org/pkg/errors/.
The blog post at https://blog.golang.org/go1.13-errors describes all the origins sentitel errors.